Build failure: b2sum

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix build github:NixOS/nixpkgs/nixos-23.11#b2sum`

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/wvsdd7wh93mczavy1bzlyafkjwjqpvwd-source
source root is source/b2sum
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: SHELL=/nix/store/0ly5lpwanagc5qpmnqrw8aqxb4ivw7kf-bash-5.2-p15/bin/bash
clang b2sum.c ../sse/blake2b.c ../sse/blake2s.c ../sse/blake2bp.c ../sse/blake2sp.c -O3 -march=native -std=c89 -Wall -Wextra -pedantic -Wno-long-long -I../sse -fopenmp  -o b2sum
b2sum.c:258:3: warning: '_Bool' is a C99 extension [-Wc99-extensions]
  bool bsdstyle = false;
  ^
/nix/store/qapnb6qpbj54m3k0ikffd4aawamc12n9-clang-wrapper-16.0.6/resource-root/include/stdbool.h:20:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
1 warning generated.
In file included from ../sse/blake2bp.c:22:
/nix/store/wdlkbj743px8j80n1ra91pv0dyrr2kh7-openmp-16.0.6-dev/include/omp.h:489:12: error: unknown type name 'inline'
    static inline int omp_is_initial_device(void) { return 1; }
           ^
1 error generated.
In file included from ../sse/blake2sp.c:21:
/nix/store/wdlkbj743px8j80n1ra91pv0dyrr2kh7-openmp-16.0.6-dev/include/omp.h:489:12: error: unknown type name 'inline'
    static inline int omp_is_initial_device(void) { return 1; }
           ^
1 error generated.
make: *** [makefile:15: all] Error 1
```

### Additional context

I wondered if this is caused by `-std=c89`, but that doesn't look to be nixpkgs-specific.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@kirelagin 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(gmacon3): `"nixpkgs-22.11-darwin"`
 - channels(root): `"nixpkgs-23.05-darwin"`
 - nixpkgs: `/Users/gmacon3/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/security/b2sum/default.nix
+++ b/pkgs/tools/security/b2sum/default.nix
@@ -23,6 +23,11 @@ stdenv.mkDerivation (finalAttrs: {
   buildInputs = [ openmp ];
 
   buildFlags = [ (lib.optional (openmp == null) "NO_OPENMP=1") ];
+
+  # clang builds require at least C99 or the build fails with:
+  # error: unknown type name 'inline'
+  env.NIX_CFLAGS_COMPILE = "-std=c99";
+
   installFlags = [ "PREFIX=$(out)" ];
 
   meta = with lib; {

